src/array.rs

```diff
@@ -0,0 +1,456 @@
+use std::ffi::c_void;
+
+use half::{bf16, f16};
+use mlx_sys::mlx_array;
+use num_complex::Complex;
+
+use crate::sealed::Sealed;
+
+// TODO: camel case?
+// Not using Complex64 because `num_complex::Complex64` is actually Complex<f64>
+#[allow(non_camel_case_types)]
+pub type c64 = Complex<f32>;
+
+/// Array element type
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u32)]
+pub enum Dtype {
```

Comment:
The organization for now is just following the c header files, but I agree that it would be better to put this in a separate mod. Could be `crate::array::dtype::Dtype`

Reply:
Moved `Dtype` to `crate::dtype::Dtype`